### PR TITLE
update to PHPStan 1.0 and fix new errors

### PR DESCRIPTION
**Pull request by LordSimal**
PHPStan 1.0 released today and therefore this PR updates the dependency for that.
https://phpstan.org/blog/phpstan-1-0-released

Also I fixed the new errors it threw due to the update.

### CODE REVIEW

File: tests/test_plugin/Model/Table/HelloWorld101010ArticlesTable.php [hunk 1]
```diff
@@ -35,15 +35,10 @@ public function clearArticle($article)
     }
 
     /**
-     * @return \Cake\Datasource\EntityInterface
+     * @return \Cake\Datasource\EntityInterface|array<\Cake\Datasource\EntityInterface>
```
Comment (dereuromark) [inline tests/test_plugin/Model/Table/HelloWorld101010ArticlesTable.php:38]: How does this return an array of entities?

Comment (LordSimal) [reply tests/test_plugin/Model/Table/HelloWorld101010ArticlesTable.php:38]: https://github.com/cakephp/cakephp/blob/4.x/src/Datasource/QueryTrait.php#L501

It didn't work with just defining
```
@return \Cake\Datasource\EntityInterface|array
```

Comment (dereuromark) [reply tests/test_plugin/Model/Table/HelloWorld101010ArticlesTable.php:38]: But you misunderstand the array, it means `array<string, mixed>` never multiple entities!

Comment (LordSimal) [reply tests/test_plugin/Model/Table/HelloWorld101010ArticlesTable.php:38]: indeed, I didn't know that. let me fix that.

**Review by dereuromark — APPROVED** [review]

**Comment by steinkel** [comment]
thank you!

**Comment by dereuromark** [comment]
You didnt want to first make the patch release for old one?

**Comment by steinkel** [comment]
You mean https://github.com/CakeDC/cakephp-phpstan/pull/4/commits/95648ecdf4761661bf5a600b2be6de8035d1d801 applied as patch 1.0.2?

**Comment by dereuromark** [comment]
No, what you already merged in master before this one, and was an important patch on top of 0.12 .

**Pull request merged**